cmd/kubeadm/app/cmd/phases/upgrade/apply/bootstraptoken.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package apply implements phases of 'kubeadm upgrade apply'.
+package apply
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	errorsutil "k8s.io/apimachinery/pkg/util/errors"
+
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/options"
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow"
+	clusterinfophase "k8s.io/kubernetes/cmd/kubeadm/app/phases/bootstraptoken/clusterinfo"
+	nodebootstraptoken "k8s.io/kubernetes/cmd/kubeadm/app/phases/bootstraptoken/node"
+)
+
+// NewBootstrapTokenPhase returns the phase to bootstrapToken
+func NewBootstrapTokenPhase() workflow.Phase {
+	return workflow.Phase{
+		Name:    "bootstrap-token",
+		Aliases: []string{"bootstraptoken"},
+		Short:   "Generates bootstrap tokens used to join a node to a cluster",
+		InheritFlags: []string{
+			options.CfgPath,
+			options.KubeconfigPath,
+			options.DryRun,
+		},
+		Run: runBootstrapToken,
+	}
+}
+
+func runBootstrapToken(c workflow.RunData) error {
+	data, ok := c.(Data)
+	if !ok {
+		return errors.New("bootstrap-token phase invoked with an invalid data struct")
+	}
+
+	if data.DryRun() {
+		fmt.Println("[dryrun] Would config cluster-info ConfigMap, RBAC Roles")
+		return nil
+	}
+
+	fmt.Println("[bootstrap-token] Configuring cluster-info ConfigMap, RBAC Roles")
+
+	client := data.Client()
+
+	var errs []error
+	// Create RBAC rules that makes the bootstrap tokens able to get nodes
+	if err := nodebootstraptoken.AllowBoostrapTokensToGetNodes(client); err != nil {
```

Comment:
may be a typo : ) s/AllowBoostrap../AllowBootstrap..

Reply:
thanks @humblec 
Good catch